lib/phoenix/endpoint/watcher.ex

```diff
@@ -2,46 +2,48 @@ defmodule Phoenix.Endpoint.Watcher do
   @moduledoc false
   require Logger
 
-  def start_link(root, cmd, args) do
-    Task.start_link(__MODULE__, :watch, [root, to_string(cmd), args])
+  def start_link(cmd, args, opts) do
+    Task.start_link(__MODULE__, :watch, [to_string(cmd), args, opts])
   end
 
-  def watch(root, cmd, args) do
-    :ok = validate(root, cmd, args)
+  def watch(cmd, args, opts) do
+    merged_opts = Keyword.merge(
+      [cd: File.cwd!(), into: IO.stream(:stdio, :line), stderr_to_stdout: true], opts)
```

Comment:
I am not sure if we can run into issues with the `File.cwd!` here. Maybe if you need it later on, you can call `inspect opts[:cd] || File.cwd!`?